lib/vehicle_factory.rb

```diff
@@ -0,0 +1,109 @@
+class VehicleFactory
+  attr_reader :vehicles
+  
+  def initialize
+    @vehicles = []
+  end
+
+  def create_wa_vehicles(dmv_data)
+    dmv_data.map do |registration|
+      vehicles << Vehicle.new({
+        vin: registration[:vin_1_10], 
+        year: registration[:model_year].to_i, 
+        make: registration[:make],
+        model: registration[:model], 
+        engine: :ev
+      })
+    end.flatten!
```

Comment:
Again, we don't need the map here since you're just shoveling all the vehicle objects into the @vehicles array. To get the proper return value, you could just call `@vehicles` on the last line of this method, outside of the map (except I'd use each) :)